pkg/cli/image/mirror/mirror.go

```diff
@@ -586,6 +586,9 @@ func (o *MirrorImageOptions) plan() (*plan, error) {
 								mustCopyLayers = true
 							case src.ref.EqualRegistry(dst.ref) && canonicalFrom.String() == canonicalTo.String():
 								// if the source and destination repos are the same, we don't need to copy layers unless forced
+							case dst.ref.Type != imagesource.DestinationRegistry:
```

Comment:
Does the same hold for case where the destination is s3?

Reply:
I haven't tested with S3, but I think so, which is why I put the test in this way.

My logic: S3 will not check if the layers for the manifest have already been uploaded, so you can end up in a situation where there is a manifest missing layers, so checking manifest existence doesn't obviate the need to check layer existence.